### ISSUE TITLE
Extended Filter Window

### ISSUE THREAD

**Issue by sznattila** [FacilMap/facilmap2#91]
Hi,
Would it be possible to have an extended filter window, where all the prop(data,"") elements can be controlled in a more user-friendly way? 
And/Or to have an option to customize the filter window by adding and removing  prop(data,"") elements to it?

**Comment by cdauth** [comment]
That would be possible.

The legend already acts as a user-friendly interface to the filters, although its functionality is quite limited (it can only filter based on dropdown/checkbox fields and types).

Actually I only added the filter window because I needed the functionality for the legend and thought I would allow an advanced usage if the functionality was already there. I really rarely use the filter function, so this would not be a very high priority for me (even though it’s probably not that complicated to implement it).

**Comment by sznattila** [comment]
We use the filters quite often so it would come handy :)

**Comment by cdauth** [comment]
What exactly are you using them for? Are you filtering based on text fields?

**Comment by sznattila** [comment]
Yep.
All of our checkboxes and dropdowns can be considered as text.
prop(data, "mycheckboxtext") == 1 and (prop(data, "mydropdown") == "secondvalue") or (prop(data, "mydropdown") == "thirdvalue")

**Comment by cdauth** [comment]
So you are only filtering against dropdowns and checkboxes?

I guess those dropdowns and checkboxes don't set the object styles, otherwise you would see them in the legend and could just click on them to change the filters.

Maybe an easier way to make this work for you would be if I would add an option for an extended mode of the legend where also items are displayed that are not connected to the object style.

**Comment by sznattila** [comment]
Yes, we do filter only against dropdowns and checkboxes.
But we have many of them and we cannot connect all of them to object style.
The idea is awesome to have an extended mode of legend for those items or to have somekind of option in the custom marker properties or anywhere to choose which items shall appear in the legend.

**Comment by cdauth** [comment]
Would it make sense for you if those items would permanently appear in the legend, even though they are not connected to a style? Or would it make more sense to have a button in the legend to expand it temporarily, just so that you can modify the filters there?

(By the way, I would be curious to know what you are using FacilMap for.)

**Comment by cdauth** [comment]
You can write me to cdauth@cdauth.eu

**Comment by georg-d** [comment]
I guess it would be relatively low effort for a noticable user experience improvement (lower the barrier & speeding up things & avoiding errors/typos) if in filter window, the lower part was not only readable documentation but also active/clickable, especially
* clicking one of the listed typeIDs like e.g. 18148 in my screenshot would add ` and typeId == 18148 ` to filter text (I added one space in front and after the text by intention)
* In line _prop(data, <field>)_ in cell _Description_ a drop down list of all filed names of all types would exist and choosing one entry would add ` and lower(prop(data, "<chosenFiledName>")) ~= "AddYourDesiredValueHere" (I added one space in front and after the text by intention)`

Screenshot with marking from https://facilmap.org/GeorgsTestA#16/47.9755/7.8661 (replace A with RW and W for different access rights)
![grafik](https://github.com/FacilMap/facilmap/assets/45993293/20060547-d464-4602-9455-d5c582078d2b)

**Comment by cdauth** [comment]
Good idea! It could also be an option to have autocompletion (and maybe even syntax highlighting) for the filter field, although I would have to investigate how much effort that is.
